builds/azure-pipelines/build-pr.yml

```diff
@@ -4,7 +4,6 @@ pr:
   branches:
     include:
     - main
-    - release/*
```

Comment:
Let's actually keep this, it's possible we'll want to have release branches in the future and keeping this doesn't really hurt anything.